Lib/test/test_ssl.py

```diff
@@ -989,6 +989,12 @@ def test_load_verify_cadata(self):
 
 
     def test_load_dh_params(self):
+        filename = u'dhpäräm.pem'
+        try:
+            filename.decode()
```

Comment:
Decoding an Unicode string doesn't make any sense :-) Moreover, you have to specify the encoding. The default encoding is ASCII, whereas the question here is if the string can be encoded to sys.getfilesystemencoding().

I propose:

fs_encoding = getfilesystemencoding()
try:
   filename.encode(fs_encoding)
except UnicodeEncodeError:
  self.skipTest("filename %r cannot be encoded to the filesystem encoding %r" % (filename, fs_encoding))

It would provide better skip message to check if the test is skipped for good reasons, or not ;-)

Reply:
Thanks for the feedback! I understand now why decoding does not make any sense for this test. Amending my last commit. :)